src/log4net/Repository/Hierarchy/Hierarchy.cs

```diff
@@ -203,22 +203,22 @@ public Logger Root
     {
       get
       {
-        if (m_root is null)
+        if (rootLogger is null)
         {
           lock (this)
```

Comment:
i know this is the existing code, but `lock(this)` is potentially problematic - something else could lock `this` from the outside; I'd recommend adding a lock object and locking that; probably ok though - it's probably been this way since the beginning.

Reply:
Thanks. I resolved it with Interlocked.CompareExchange.